scripts/install_dependencies/install_openvino_dependencies.sh

```diff
@@ -128,7 +128,7 @@ elif [ "$os" == "ubuntu18.04" ] ; then
     pkgs_core=(libtbb2 libpugixml1v5)
     pkgs_gpu=(ocl-icd-libopencl1)
     pkgs_python=(python3.8 libpython3.8 python3.8-venv python3-pip)
-    pkgs_dev=(cmake pkg-config g++ gcc libc6-dev libgflags-dev zlib1g-dev nlohmann-json-dev make curl sudo)
+    pkgs_dev=(cmake pkg-config g++ gcc libc6-dev libgflags-dev zlib1g-dev nlohmann-json-dev make curl sudo libnuma-dev)
```

Comment:
why do we need it?

Reply:
We want to precisely control the memory to be allocated on specific numa node to balance the memory access cost. The needed function is in libnuma-dev package.

Reply:
on what platforms do we have such dependency? 
is it possible to use hwloc, which is already OpenVINO dependency?

Reply:
Currently will be in Linux SPR. The dependency of hwloc is its binary, we need add the dev package `libhwloc-dev` if preferring hwloc. Not sure the `libhwloc-dev` could be used instead of `libnuma-dev`. Do we have a PR for reference, which includes all places to be added for a new development package?

Reply:
When you are adding dependency, you should not depend on specific package. Otherwise, how are we going to use such dependency in pip package? we cannot depend on installation via apt

I suppose we need to add such dependency as a submodule first. As a second step, we need to find such dependency in other package managers (as I wrote, on Linux we support Conan, brew, vcpkg, apt, yum, pip package managers, compilation from sources)

Reply:
Change to use dlopen/dlsym to dynamic load `libnuma.so` to avoid the compilation stage dependency.

Reply:
what difference will users see if they install / don't install this dependency on Linux SPR?

Reply:
if don't install, the numa-aware allocation will be fallback to normal malloc. Even on oneTBB [official website](https://oneapi-src.github.io/oneTBB/index.html) there is also no information about tbbbind, so not sure if it can be used.